### ISSUE TITLE
Navigating between Multichannel and Stereo Layers of one Disc

### ISSUE THREAD

**Issue by Reemtsman** [xbmc/audiodecoder.sacd#16]
All SACD´s have a Stereo Layer, therefore using the Stereo Preset is a good starting point.

But if you want to take full advantage of your Multichanned SACD´s either the "All" or the "Multichannel" Preset must be used.
- Multichanned works well with MCH-SACD´s but it leads to a strange behaviour when Stereo-Only-SACD´s are played. The Stereo Track is now played as 4.0 (...or 4.1) instead of 2.0 PCM.
- All leads to another strange behaviour because both Stereo and MCH Tracks are played after another when playing the Album:
1. Track 1 Stereo
2. Track 1 MCH
3. Track 2 Stereo
4. Track 2 MCH
...
Here a smarter approach to navigation is needed. Either a modification of the "All" Preset that allows to choose between Stereo or MCH when starting a Album.
Or a modification of the MCH preset that makes sure that 2.0 is not pumped up to 4.0 when playing Stereo-only SACD´s

**Comment by fritsch** [comment]
Reason for this (it's also documented in the wiki):

Many users have playlists with changing content and maybe an AVR that does a "burrrrrtsch" noise, whenever it switches formats, therefore kodi has a default "behaviour" that is called "Optimized" see the PCM Music configuration, means: if you play a playlist with 5.1 file and afterwards a 2.0 file is coming, kodi keeps the 5.1 configuration and plays the upcoming 2.0 with 6 channels, where FL, FR have the sound and the others are filled with zeroes. This is especially handy when you watch TV and the movie switches to commercials (stereo).

In short: The setting "Best Match" will reconfigure the audio sink whenever the stream changes. Choose "Best match".

**Comment by Reemtsman** [comment]
Thank you fritsch. Best Match is the Setting that i am already using. But i cannot believe it is intended, that Kodi throws the tracks from Stereo Layer into one playlist together with the MCH Tracks. Nobody wants to hear the same song 2 times after another, first in stereo and then again in MCH!
When i play an album from a SACD Iso i want tho have the choice to play the whole album in Stereo or the whole album in MCH! And this selection should be possible without going into the settings of the Add-on.

**Comment by fritsch** [comment]
PullRequest welcome. As you might have realized by now: Kodi is no company, kodi is driven by free time developers, which might have lost interest in the middle after the initial change.

Therefore: As you have a great value from kodi -> consider contributing and fixing the rough edges. But take care with "no one" :-) you will most likely find a very loud-speaking guy which's use-case you will kill.

**Comment by Reemtsman** [comment]
@fritsch : Sorry for my general statement on Nobody/Everybody! Your are perfectly right, using absolute terms almost imideately turns you into a liar.
I am well aware that Kodi is not a giant Company with unlimited resources, therefore i am awfully greatfull and heavily impressed by everything you have achieved with this amazing App!
Unfortunately i am just a enduser with zero IT skills. I can only support the continuous improvement of Kodi through reporting issues but i cannot help fixing the rough edges.

**Comment by AlwinEsch** [comment]
New versio 19.0.0 now released!

Since you already have it and the setting is therefore on "All", you have to set it to "Stereo" or "Multichannel" here. Should then only be there once the individual songs.
![Bildschirmfoto vom 2021-09-30 13-30-39](https://user-images.githubusercontent.com/6879739/135447742-0ec21dea-bb03-41af-b08c-372c162c7685.png)

Related to;
> Thank you fritsch. Best Match is the Setting that i am already using. But i cannot believe it is intended, that Kodi throws the tracks from Stereo Layer into one playlist together with the MCH Tracks. Nobody wants to hear the same song 2 times after another, first in stereo and then again in MCH! When i play an album from a SACD Iso i want tho have the choice to play the whole album in Stereo or the whole album in MCH! And this selection should be possible without going into the settings of the Add-on.

Handling has been improved in the new released version.

Once there is a new option in add-on settings to split the list, whereby all stereo and then all multi-channel songs are listed.
![Bildschirmfoto vom 2021-09-30 13-34-08](https://user-images.githubusercontent.com/6879739/135447991-6788e9c7-7b0e-43c2-85fa-592727bf7308.png)

In addition, the handling of the "Preferable area" setting has been improved, e.g. if stereo is selected and only multi-channel is available, this is used instead, the same also in the other way. Therefore the setting "Separate ..." is deactivated.
![Bildschirmfoto vom 2021-09-30 13-34-40](https://user-images.githubusercontent.com/6879739/135448050-e710d9b9-9a27-43bf-9471-955fb3699b57.png)


Therefore this setting is now also with "Multichannel" as default.  And issue here becomes closed.

--------------

Related to;
> Reason for this (it's also documented in the wiki):
> 
> Many users have playlists with changing content and maybe an AVR that does a "burrrrrtsch" noise, whenever it switches formats, therefore kodi has a default "behaviour" that is called "Optimized" see the PCM Music configuration, means: if you play a playlist with 5.1 file and afterwards a 2.0 file is coming, kodi keeps the 5.1 configuration and plays the upcoming 2.0 with 6 channels, where FL, FR have the sound and the others are filled with zeroes. This is especially handy when you watch TV and the movie switches to commercials (stereo).
> 
> In short: The setting "Best Match" will reconfigure the audio sink whenever the stream changes. Choose "Best match".

However, this change here on addon has no effect on Kodi's own changeover between switched audio channels for output.

**Comment by Reemtsman** [comment]
.....Amazing! Du bist der Beste!! Or: You are the best! (for the ROW;-)

But i cannot see a updated Version 19.0.0 of the Add-on in the Kodi-Repo. This still displays v 0.1.2.

**Comment by Reemtsman** [comment]
...i am confused: Are you Referring to v19.0.0 of MATRIX??

This looks like a Version Step backwards as the latest stable version was already 19.1

**Comment by AlwinEsch** [comment]
Only the major version should be equal, the minor (now the 0) is now more or less the major for addon beginning with 0 as 1.

> But i cannot see a updated Version 19.0.0 of the Add-on in the Kodi-Repo. This still displays v 0.1.2.

It takes a bit until them present on our mirrors and his addons.xml.gz where used for updates, see http://mirrors.kodi.tv/addons/matrix/audiodecoder.sacd+windows-x86_64/

**Comment by Reemtsman** [comment]
....confirmed! Now 19.0.0 is displayed!
Hast the Problem with the Video-Iso-Conflict been solved in this version as well?

**Comment by AlwinEsch** [comment]
The problem has been found and the associated request created, but not yet fixed in Kodi. Wait for a confirm on this request https://github.com/xbmc/xbmc/pull/20205, and it will probably only come into Kodi 19.3 as backward.

If you can create Kodi yourself, you could take the patch of the request to fix the problem earlier.

**Comment by AlwinEsch** [comment]
Releated to this issue. It just comes into my head not to take multichannel as default :thinking: , but to leave it in stereo. It's cool, but don't think everyone has such powerful CPUs for it :sweat_smile:.

In 19.0.1 about addon version, I'll set it to be stereo as the default.
